src/System.CommandLine.Subsystems.Tests/HelpSubsystemTests.cs

```diff
@@ -0,0 +1,89 @@
+
+using FluentAssertions;
+using Xunit;
+using System.CommandLine.Parsing;
+
+namespace System.CommandLine.Subsystems.Tests
+{
+    public class HelpSubsystemTests
+    {
+        [Fact]
+        public void When_help_subsystem_is_used_the_help_option_is_added_to_each_command_in_the_tree()
+        {
+            var rootCommand = new CliRootCommand
+                {
+                    new CliOption<bool>("-x"), // add option that is expected for the test data used here
+                    new CliCommand("a")
+                };
+            var configuration = new CliConfiguration(rootCommand);
+
+            var pipeline = Pipeline.CreateEmpty();
+            pipeline.Help = new HelpSubsystem();
+
+            // Parse is used because directly calling Initialize would be unusual
+            var result = pipeline.Parse(configuration, "");
+
+            rootCommand.Options.Should().NotBeNull();
+            rootCommand.Options
+                .Count(x => x.Name == "--help")
+                .Should()
+                .Be(1);
+            var subcommand = rootCommand.Subcommands.First();
+            subcommand.Options.Should().NotBeNull();
+            subcommand.Options
+                .Count(x => x.Name == "--help")
+                .Should()
+                .Be(1);
+        }
+
+        [Theory]
+        [ClassData(typeof(TestData.Help))]
+        public void Help_is_activated_only_when_requested(string input, bool result)
+        {
+            CliRootCommand rootCommand = [new CliOption<bool>("-x")]; // add random option as empty CLIs are rare
+            var configuration = new CliConfiguration(rootCommand);
+            var helpSubsystem = new HelpSubsystem();
+            var args = CliParser.SplitCommandLine(input).ToList().AsReadOnly();
+
+            Subsystem.Initialize(helpSubsystem, configuration, args);
+
+            var parseResult = CliParser.Parse(rootCommand, input, configuration);
+            var isActive = Subsystem.GetIsActivated(helpSubsystem, parseResult);
+
+            isActive.Should().Be(result);
+        }
+
+        [Fact]
+        public void Outputs_help_message()
+        {
+            var consoleHack = new ConsoleHack().RedirectToBuffer(true);
+            var helpSubsystem = new HelpSubsystem();
+            Subsystem.Execute(helpSubsystem, new PipelineResult(null, "", null, consoleHack));
+            consoleHack.GetBuffer().Trim().Should().Be("Help me!");
+        }
+
+        [Fact]
+        public void Custom_help_subsystem_can_be_used()
+        {
+            var consoleHack = new ConsoleHack().RedirectToBuffer(true);
+            var pipeline = Pipeline.CreateEmpty();
+            pipeline.Help = new AlternateSubsystems.AlternateHelp();
+
+            pipeline.Execute(new CliConfiguration(new CliRootCommand()), "-h", consoleHack);
+            consoleHack.GetBuffer().Trim().Should().Be($"***Help me!***");
```

Comment:
I might be misssing something but it seems like `Custom_help_subsystem_can_be_used` and `Custom_help_subsystem_can _replace_standard` are the same test. If so I think we might want to consolidate them

Reply:
I've copied these shamelessly from [VersionSubsystemTests.cs](https://github.com/dotnet/command-line-api/blob/main-powderhouse/src/System.CommandLine.Subsystems.Tests/VersionSubsystemTests.cs#L96)

Couldn't really tell why there are two tests with identical body there, but I assumed I don't understand something and chose to add it. Removing it in the upcoming commit. Should I also remove the duplicate tests from `VersionSubsystemTests.cs` - specifically `Custom_version_subsystem_can_replace_standard`?